narwhals/_arrow/series.py

```diff
@@ -584,6 +584,162 @@ def to_string(self, format: str) -> ArrowSeries:  # noqa: A002
             pc.strftime(self._arrow_series._native_series, format)
         )
 
+    def year(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.year(self._arrow_series._native_series)
+        )
+
+    def month(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.month(self._arrow_series._native_series)
+        )
+
+    def day(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.day(self._arrow_series._native_series)
+        )
+
+    def hour(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.hour(self._arrow_series._native_series)
+        )
+
+    def minute(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.minute(self._arrow_series._native_series)
+        )
+
+    def second(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.second(self._arrow_series._native_series)
+        )
+
+    def millisecond(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.millisecond(self._arrow_series._native_series)
+        )
+
+    def microsecond(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        result = pc.add(pc.multiply(pc.millisecond(arr), 1000), pc.microsecond(arr))
+
+        return self._arrow_series._from_native_series(result)
+
+    def nanosecond(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        pc = get_pyarrow_compute()
+        arr = self._arrow_series._native_series
+        result = pc.add(
+            pc.multiply(self.microsecond()._native_series, 1000), pc.nanosecond(arr)
+        )
+        return self._arrow_series._from_native_series(result)
+
+    def ordinal_day(self: Self) -> ArrowSeries:
+        pc = get_pyarrow_compute()
+        return self._arrow_series._from_native_series(
+            pc.day_of_year(self._arrow_series._native_series)
+        )
+
+    def total_minutes(self: Self) -> ArrowSeries:
```

Comment:
For all the `total_*` methods, I am using the following strategy:

- Extract array time unit (question, should we add more?)
- Convert time unit to factor of `total_*` unit
- Divide or multiply by such scalar: notice that duration cannot be multiplied/divided by float/double, hence we need to distinguish for when to multiply and when to divide
- Finally cast to int: this is because polars returns int dtype

Reply:
Edit: Possible time units for pyarrow duration are "s", "ms", "us", "ns". Now all these are covered :)